### DISCUSSION TITLE
Calling Enterprise customers! Try out the new auto-add workflow.

### DISCUSSION THREAD

**Discussion by 2percentsilk** [community/community#44816] [Enterprise]
### Select Topic Area

Show & Tell

### Body

We've added the ability to set up a workflow to automatically add items to your project. Check out [our docs](https://docs.github.com/en/enterprise-cloud@latest/issues/planning-and-tracking-with-projects/automating-your-project/adding-items-automatically) to learn more about configuring the workflow and try it out today!

**Comment by tay1orjones** [comment]
This is great! I'm glad to see this added to the workflows UI. It looks like for now this is limited to a single repo. Multi-repo support would be really beneficial for us. 

We have issues from 3-4 different repos added to a single project view for triage. Right now we're doing this by having a reusable workflow `.yml` file across the different repos.

**Reply by 2percentsilk** [reply]
@tay1orjones multi-repo support is in the works! :D just hang tight :)

**Reply by aomarks** [reply]
Would this include automatically adding issues from _external_ repos (meaning repos in other orgs)? That would be huge for us, since our team manages many repos across many orgs, but we want everything in a single org's project.

**Reply by john-craft** [reply]
Definitely need multi-repo support!

**Reply by cnordvik** [reply]
> @tay1orjones multi-repo support is in the works! :D just hang tight :)

Any ETA on it? Just started 30-day trial to see if we can use this feature instead of Zenhub which we currently use for tracking all issues in a single board across 8 repositories 😅

**Reply by tay1orjones** [reply]
Excited to see this [shipped](https://github.blog/changelog/2023-02-23-github-issues-projects-february-23rd-update/) today 🎉  Thanks!

**Reply by cnordvik** [reply]
Any chance to increase the limit from 4 to 5 or 10? We have each client in it's own repos + design so we need 5 workflows and the current limit is 4 😅 

![image](https://user-images.githubusercontent.com/584549/221131758-bbfcdb05-56bc-43e0-af5f-67dab426aebd.png)

**Reply by rileybroughten** [reply]
We've increased the limit for enterprise users to 20 total auto-add workflows, check out the announcement [here](https://github.blog/changelog/2023-03-09-github-issues-projects-march-9th-update/#robot-automatically-add-and-archive-project-items) as well as [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/adding-items-automatically) for more details 🎉

**Comment by lenzai** [comment]
Come on. Add it to the ~~basic~~ teams plan please!

**Reply by lfrancke** [reply]
What is disappointing about this decision is that for over a year we've been told "it's coming soon" (see my [original thread](https://github.com/orgs/community/discussions/5378) from August 2021).
Never once has there been a mention of it being Enterprise only. Had there been at least we would have worked on alternatives in the meantime.

The decision is your's and it's fine for me in general, we get a lot of value for free out of Github. I'd urge you to consider announcing plans like this in advance though as that decision surely wasn't made just yesterday.

Edit: We even are on an Enterprise plan at the moment

**Reply by lenzai** [reply]
sure free plan is already feature rich.
But I don't see any reason to exclude this feature from paid plan. Excluding "teams" plan really does not make sense.

**Reply by christofferbredberg** [reply]
Would also love to see this for teams plan!

**Reply by marina-chibizova** [reply]
+1

**Reply by rileybroughten** [reply]
Auto-add workflows have been released to all users! Check out the announcement [here](https://github.blog/changelog/2023-03-09-github-issues-projects-march-9th-update/#robot-automatically-add-and-archive-project-items) as well as [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/adding-items-automatically) for more details, as Team users can have up to 5 auto-add workflows 🎉

**Comment by heitorlessa** [comment]
I've just tested and thank you for this qualify of life improvement - I was already considering other options.

It'd be incredibly useful if this feature could:

* Set custom field values when a rule matches: e.g., `is:open label:triage` -> Add to the project -> set `Status` to `Pending Review`
* Ability to create multiple rules so we're not limited to adding `triage` issues to the board only
* Ability to work with more than one repository (nice-to-have): e.g., `repository: X`

**Reply by 2percentsilk** [reply]
Is status the only thing you'd want to set as part of auto-add?

**Reply by debonte** [reply]
Sprint would be nice

**Comment by harvinders** [comment]
Another thing we would really appreciate, apart from extending this to multiple repositories, is an ability to move the issues to `In progress` when a PR is opened with linked issues.

**Reply by john-craft** [reply]
Yes, absolutely!

**Reply by 2percentsilk** [reply]
I know PR based triggers would be great for folks!

**Reply by debonte** [reply]
Similarly, being able to move issues to `Done` when the associated PR is closed would be nice.

**Reply by john-craft** [reply]
> Similarly, being able to move issues to `Done` when the associated PR is closed would be nice.

That should work indirectly already if the PR is associated with the issue. PR merge => close issue => issue marked "Done" on project.

**Reply by debonte** [reply]
Makes sense. My team doesn't close issues until the fix ships, but we move the issues to the Done column as soon as the fix is merged. Maybe we're just weird?

**Reply by ashb** [reply]
With things like Dependabot automatically opening PRs this is very annoying to deal with.

**Reply by humphreyja** [reply]
Agreed 100% with this. For our team, we have support staff that create issues labeled "bug" and would like to know if it is getting done without needing to keep reaching out to our developers on progress updates. A simple workflow called `Issue Changed` with a condition `when PR Assigned` do `Set Status: In Progress` would be awesome.

A step further, if there was the ability to add multiple of those, like `Issue Changed` when `assigned to user` do `Set Status: Investigating`.

**Comment by yngvark** [comment]
Last time i tested, I was able to use this workflow on maximum 4 repositories, this limitation is now increased or removed - great!

**Comment by d-i-bondarenko** [comment]
I've made a workflow for auto-add closed issues to project and it doesn't work. 

`is:issue is:closed`

It shows filter but freshly closed issues don't auto-add to my "graveyard" project 😁

**Reply by rileybroughten** [reply]
@d-i-bondarenko is this issue still occurring for you?

**Reply by maiconsanson** [reply]
Hi @rileybroughten, I have a similar issue with the **Auto-add to project**.

When I try to add any status from an `issue` or `pr`, the filter doesn't match at all.

<img width="679" alt="Screenshot 2023-06-30 at 15 51 30" src="https://github.com/community/community/assets/1965455/0595359d-bd2f-4e47-972a-41a853065614">

I've checked the generated query to filter and seems to be wrong:

`-project:maiconsanson/1 is:open,closed`.

<img width="482" alt="Screenshot 2023-06-30 at 15 52 35" src="https://github.com/community/community/assets/1965455/a3c73a77-f329-4890-bb68-ef3109f1fe06">

Any thoughts?

**Comment by caseyokane-8451** [comment]
Any possibility of making these workflows configurable/editable via the REST API?

Also, this is probably a bit greedy but any reason why the limit is for 20 (and any chance it might be increased eventually)? Our team is trying to manage our Roadamap project (which includes 30-40 repos).

**Comment by quartercastle** [comment]
Is there any plans for increasing the limit of 20 repositories? We manage around 40-50 repositories and it would be nice if we could have them all configured to auto-add.

**Comment by oddsund** [comment]
Adding to the call for higher limit + api-configuration support.

We have more than 20 repos we want to join into one project, and while a higher limit would help to get this done, it would still be a pain to manage via clickety-clicks.

I see two options today; Github Actions Workflow or Webhook configured in settings.
For Github Action, each repository needs a Github workflow with either a PAT as a secret or a Github App secret. Neither is especially desired because of the requirement of managing the same workflow across multiple repositories, and keeping the same secret in sync across multiple repositories.

For the webhook, an external service could handle adding issues to a project, but that's more code to main, for a feature that's "almost there". Also, we have to manage a webhook across multiple repositories.

So yeah.. Higher limit + api-configuration support would be greatly appreciated 🙏

**Comment by CoderNumber1** [comment]
I'm in a similar boat to others that have commented. We would benefit greatly from increasing the auto-add workflow limit beyond 20 repositories.
